node/src/effect.rs

```diff
@@ -1890,78 +1900,38 @@ impl<REv> EffectBuilder<REv> {
         .await
     }
 
-    /// Retrieves an `Account` from global state if present.
-    pub(crate) async fn get_account_from_global_state(
+    /// Retrieves an `AddressableEntity` from under the given key in global state if present.
+    pub(crate) async fn get_addressable_entity(
         self,
         state_root_hash: Digest,
-        account_key: Key,
-    ) -> Option<StoredValue>
+        key: Key,
+    ) -> Option<AddressableEntity>
     where
         REv: From<ContractRuntimeRequest>,
     {
-        let query_request = QueryRequest::new(state_root_hash, account_key, vec![]);
-        match self.query_global_state(query_request).await {
-            Ok(QueryResult::Success { value, .. }) => Some(*value),
-            Ok(_) | Err(_) => None,
-        }
-    }
-
-    /// Retrieves the balance of a purse, returns `None` if no purse is present.
-    pub(crate) async fn check_purse_balance(
-        self,
-        state_root_hash: Digest,
-        main_purse: URef,
-    ) -> Option<U512>
-    where
-        REv: From<ContractRuntimeRequest>,
-    {
-        let balance_request = BalanceRequest::new(state_root_hash, main_purse);
-        match self.get_balance(balance_request).await {
-            Ok(balance_result) => {
-                if let Some(motes) = balance_result.motes() {
-                    return Some(*motes);
-                }
-                None
-            }
-            Err(_) => None,
-        }
-    }
-
-    /// Retrieves an `Contract` from global state if present.
-    pub(crate) async fn get_contract_for_validation(
-        self,
-        state_root_hash: Digest,
-        query_key: Key,
-        path: Vec<String>,
-    ) -> Option<Box<AddressableEntity>>
-    where
-        REv: From<ContractRuntimeRequest>,
-    {
-        let query_request = QueryRequest::new(state_root_hash, query_key, path);
-        match self.query_global_state(query_request).await {
-            Ok(QueryResult::Success { value, .. }) => {
-                // TODO: Extending `StoredValue` with an `into_contract` would reduce cloning here.
-                value.as_addressable_entity().map(|c| Box::new(c.clone()))
-            }
-            Ok(_) | Err(_) => None,
-        }
+        self.make_request(
+            |responder| ContractRuntimeRequest::GetAddressableEntity {
+                state_root_hash,
+                key,
+                responder,
+            },
+            QueueKind::ContractRuntime,
+        )
+        .await
     }
 
-    /// Retrieves an `ContractPackage` from global state if present.
-    pub(crate) async fn get_contract_package_for_validation(
+    /// Retrieves a `Package` from under the given key in global state if present.
+    pub(crate) async fn get_contract_package(
```

Comment:
```suggestion
    pub(crate) async fn get_package(
```

Reply:
@darthsiroftardis is supposed to be fixing up all the account / contract related renames in a later PR